Gems/Atom/RHI/Vulkan/Code/Source/RHI/SwapChain.cpp

```diff
@@ -256,7 +271,19 @@ namespace AZ
                 info.pImageIndices = &imageIndex;
                 info.pResults = nullptr;
 
-                const VkResult result = vkQueuePresentKHR(vulkanQueue->GetNativeQueue(), &info);
+                VkResult result = vkQueuePresentKHR(vulkanQueue->GetNativeQueue(), &info);
+
+                // Vulkan doesn't directly support waiting on a specified number of vsync intervals. Instead, emulate
+                // >1 vblank waits by presenting the same frame as necessary. e.g. m_verticalSyncInterval == 2 => 1 extra present
+                for (int i = 0; i < static_cast<int>(GetDescriptor().m_verticalSyncInterval) - 1; ++i)
```

Comment:
I'm not sure if this is correct. You are presenting the same image multiple times but using only one semaphore to know when the presentation engine has finish using it. If that semaphore is signaled after the first "present", then the engine will start overwriting the image while the presentation engine is doing the second "present".

The way it is at the moment is there are two separate semaphores per swapchan image: one that the presentation engine waits until the engine finish rendering everything to that image. This is signal by the last scope that is using that image. The other semaphore is to tell the engine when is safe to start rendering to a swapchain image. This semaphore is signal by the presentation engine when it finishes presenting the image to the screen. But if we are presenting multiple times the same image, I'm not sure when is that second semaphore is going to be signal.

Reply:
Ok I'll take a closer look. Naively I assumed that the presentation completion was done by draining the presentation queue itself instead of relying on a semaphore (at least, I didn't see this second semaphore when I looked last night)

Reply:
I wonder if there is a way to  use multiple semaphores per image (One per m_verticalSyncInterval) for when the presentation engine signals the engine. On the engine side we check if all the semaphores are signaled before trying to write into the image.

Reply:
Hmm I'm not actually sure how the interaction with the synchronous acquire would work. In particular, I'm not seeing much precedent for multiple presents without interleaved acquires.